jenkins-plugin/src/main/java/org/jenkinsci/plugins/pipeline/maven/reporters/TasksScannerReporter.java

```diff
@@ -0,0 +1,106 @@
+package org.jenkinsci.plugins.pipeline.maven.reporters;
+
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.model.Run;
+import hudson.model.StreamBuildListener;
+import hudson.model.TaskListener;
+import hudson.plugins.findbugs.FindBugsPublisher;
+import hudson.plugins.tasks.TasksPublisher;
+import org.jenkinsci.plugins.pipeline.maven.MavenSpyLogProcessor;
+import org.jenkinsci.plugins.pipeline.maven.ResultsReporter;
+import org.jenkinsci.plugins.pipeline.maven.util.XmlUtils;
+import org.jenkinsci.plugins.workflow.steps.StepContext;
+import org.w3c.dom.Element;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.annotation.Nonnull;
+
+/**
+ * @author <a href="mailto:cleclerc@cloudbees.com">Cyrille Le Clerc</a>
+ * @see hudson.plugins.tasks.TasksPublisher
+ */
+public class TasksScannerReporter implements ResultsReporter {
+    private static final Logger LOGGER = Logger.getLogger(FindbugsAnalysisReporter.class.getName());
+
+    /*
+    <ExecutionEvent type="ProjectSucceeded" class="org.apache.maven.lifecycle.internal.DefaultExecutionEvent" _time="2017-03-08 21:03:33.564">
+        <project baseDir="/Users/cleclerc/git/jenkins/pipeline-maven-plugin/maven-spy" file="/Users/cleclerc/git/jenkins/pipeline-maven-plugin/maven-spy/pom.xml" groupId="org.jenkins-ci.plugins" name="Maven Spy for the Pipeline Maven Integration Plugin" artifactId="pipeline-maven-spy" version="2.0-beta-7-SNAPSHOT">
+          <build sourceDirectory="/Users/cleclerc/git/jenkins/pipeline-maven-plugin/maven-spy/src/main/java" directory="/Users/cleclerc/git/jenkins/pipeline-maven-plugin/maven-spy/target"/>
+        </project>
+        ...
+    </ExecutionEvent>
+     */
+    @Override
+    public void process(@Nonnull StepContext context, @Nonnull Element mavenSpyLogsElt) throws IOException, InterruptedException {
+        TaskListener listener = context.get(TaskListener.class);
+        if (listener == null) {
+            LOGGER.warning("TaskListener is NULL, default to stderr");
+            listener = new StreamBuildListener((OutputStream) System.err);
+        }
+        FilePath workspace = context.get(FilePath.class); // TODO check that it's the good workspace
+        Run run = context.get(Run.class);
+        Launcher launcher = context.get(Launcher.class);
+
+
+        try {
+            Class.forName("hudson.plugins.tasks.TasksPublisher");
```

Comment:
+1 but I would first like to replace the XML DOM Document by a JAXB Object Oriented approach. See https://github.com/cyrille-leclerc/pipeline-maven-plugin/tree/jaxb-v2